src/g2index.F90

```diff
@@ -265,25 +265,25 @@ END SUBROUTINE GETG2I
 !> - 3 Error deallocating memory.
 !>
 !> @author Mark Iredell @date 1995-10-31
-SUBROUTINE GETG2IR(LUGB, MSK1, MSK2, MNUM, CBUF, NLEN, NNUM, NMESS, IRET)
-  USE RE_ALLOC              ! NEEDED FOR SUBROUTINE REALLOC
+subroutine getg2ir(lugb, msk1, msk2, mnum, cbuf, nlen, nnum, nmess, iret)
+  use re_alloc              ! needed for subroutine realloc
   implicit none
 
-  CHARACTER(LEN = 1), POINTER, DIMENSION(:) :: CBUF
-  INTEGER, INTENT(IN) :: LUGB, MSK1, MSK2, MNUM
-  INTEGER, INTENT(OUT) :: NLEN, NNUM, NMESS, IRET
-  CHARACTER(LEN = 1), POINTER, DIMENSION(:) :: CBUFTMP
+  character(len = 1), pointer, dimension(:) :: cbuf
+  integer, intent(in) :: lugb, msk1, msk2, mnum
+  integer, intent(out) :: nlen, nnum, nmess, iret
+  character(len = 1), pointer, dimension(:) :: cbuftmp
   integer :: nbytes, newsize, next, numfld, m, mbuf, lskip, lgrib
   integer :: istat, iseek, init, iret1
-  PARAMETER(INIT = 50000, NEXT = 10000)
+  parameter(init = 50000, next = 10000)
 
-  INTERFACE      ! REQUIRED FOR CBUF POINTER
-     SUBROUTINE IXGB2(LUGB, LSKIP, LGRIB, CBUF, NUMFLD, MLEN, IRET)
-       INTEGER, INTENT(IN) :: LUGB, LSKIP, LGRIB
-       CHARACTER(LEN = 1), POINTER, DIMENSION(:) :: CBUF
-       INTEGER, INTENT(OUT) :: NUMFLD, MLEN, IRET
-     END SUBROUTINE IXGB2
-  END INTERFACE
+  interface      ! required for cbuf pointer
+     subroutine ixgb2(lugb, lskip, lgrib, cbuf, numfld, mlen, iret)
+       integer :: lugb, lskip, lgrib
+       character(len = 1), pointer, dimension(:) :: cbuf
+       integer :: numfld, mlen, iret
```

Comment:
Is there a reason for removing the intent's here? `lugb` for instance is documented under `ixgb2()` as having intent 'in' but I don't see that being set anywhere.

Reply:
The intents are not in the code, and are a mistake in the documentation.

I would add them to the code but I'm not sure how that would affect in existing code. So instead I'm removing them from the documentation.